packages/react-instantsearch/src/core/utils.js

```diff
@@ -58,3 +58,6 @@ export function assertFacetDefined(searchParameters, searchResults, facet) {
 export function getDisplayName(Component) {
   return Component.displayName || Component.name || 'UnknownComponent';
 }
+
+const resolved = typeof Promise !== 'undefined' && Promise.resolve();
+export const defer = resolved ? f => { resolved.then(f); } : setTimeout;
```

Comment:
Note: on the benchmark you might get better results with other implementations but Promise.resolve() is widely supported while others are not